c2corg_ui/static/js/gpxupload.js

```diff
@@ -59,6 +61,60 @@ app.GpxUploadController = function($scope) {
 
 
 /**
+ *
+ * @param {Array.<ol.Feature>} features - invalid features that have to be checked
+ * @returns {Array.<ol.Feature>} features - this geometry is valid
+ * @private
+ */
+app.GpxUploadController.prototype.validatedFeatures_ = function(features) {
+  for (var i = 0; i < features.length; i++) {
+    /**
+     *
+     * @type {ol.geom.Geometry}
+     */
```

Comment:
In general, it is good practice to type function parameters and class members (`this.xyz`). But we usually don't do it for local variables because the compiler can figure out the type itself.